### DISCUSSION TITLE
[set -o vi] what's the easiest way to cancel the current entered command? Control-C does not work

### DISCUSSION THREAD

**Discussion by jmlucjav** [akinomyoga/ble.sh#510] [Q&A] [answered]
in my case, Control-c will offer you to exit the console instead. Is there an easy way I am missing to just start over? The usecase is when you have pasted a long multiline text, and have been editing it, so just undoing is not fast...

thanks!

**Comment by akinomyoga** [comment, marked as answer]
Control-C is supposed to work by default if you use the `emacs` editing mode. If you use the `vi` editing mode, the way to configure it is described [here](https://github.com/akinomyoga/ble.sh/wiki/Vi-%28Vim%29-editing-mode#normalinsert-mode-c-c-cancel--discard-line) and [here](https://github.com/akinomyoga/ble.sh/blob/32f290df78554e22d7ff779d34f1dfec271c16df/blerc.template#L1218-L1223).

----

To quote them,

> ## (Normal/insert mode) `C-c`: cancel / discard-line
>
> The default mapping of <kbd>C-c</kbd> is `vi_imap/normal-mode-without-insert-leave` (in `vi_imap`), `vi-command/cancel` (in `vi_nmap`).
> If you instead want to discard the current line and go to the next line, you can bind <kbd>C-c</kbd> to 'discard-line':
>
> ```
> ble-bind -m vi_imap -f 'C-c' discard-line
> ble-bind -m vi_nmap -f 'C-c' discard-line
> ```


https://github.com/akinomyoga/ble.sh/blob/32f290df78554e22d7ff779d34f1dfec271c16df/blerc.template#L1217-L1224

**Reply by jmlucjav** [reply]
thanks! have missed that
